src/test/java/core/CredentialsTest.java

```diff
@@ -1,55 +1,227 @@
+        createUpdateDeleteTest(GLOBAL_SCOPE);
+    }
+    
+    private void createUpdateDeleteTest(String systemScope) {
+        final ManagedCredentials c = new ManagedCredentials(jenkins);
+
+        // Create credential
+        c.open();
+        UserPwdCredential upc = c.add(UserPwdCredential.class);
+        upc.username.set(CRED_USER);
+        upc.password.set(CRED_PWD);
+        upc.description.set("Credential description");
+        upc.scope.select(systemScope);
+        upc.setId(CRED_ID);
+        c.save();
+
+        // verify credential was created
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+        
+        // Update credential
+        c.open();
+        upc = c.get(UserPwdCredential.class, CRED_ID);
+        final String usernameModified = CRED_USER + "-Modified";
+        upc.username.set(usernameModified);
+        c.save();
+        
+        // verify credential was updated
+        verifyInputValue(CRED_ID, USERNAME_INPUT, usernameModified);
+        
+        // Remove credential 
+        c.open();
+        upc = c.get(UserPwdCredential.class, CRED_ID);
+        upc.delete.click();
+        c.save();
+        
+        // verify credential is not present
+        verifyCredentialNotPresent(CRED_ID);
+    }
+    
+    @Test
+    public void manageDomainCredentialsTest() {
+        final String domainName = "domain";
+        // Create domain and credential inside the domain
+        final ManagedCredentials c = new ManagedCredentials(jenkins);
+        c.open();
+        Domain d = c.addDomain();
+        d.name.set(domainName);
+        d.description.set("domain description");
+        UserPwdCredential credInDomain = d.addCredential(UserPwdCredential.class);
+        credInDomain.username.set(CRED_USER);
+        credInDomain.password.set("cPwd");
+        credInDomain.description.set("cDesc");
+        credInDomain.scope.select(SYSTEM_SCOPE);
+        credInDomain.setId(CRED_ID);
+        c.save();
+        
+        verifyInputValueInDomain(domainName, CRED_ID, USERNAME_INPUT, CRED_USER);
+        
+        // Update credential inside the domain
+        c.open();
+        credInDomain = c.get(UserPwdCredential.class, domainName, CRED_ID);
+        String credUserModified = CRED_USER + "-Modified";
+        credInDomain.username.set(credUserModified);
+        c.save();
+        
+        verifyInputValueInDomain(domainName, CRED_ID, USERNAME_INPUT, credUserModified);
+        
+        // Remove credential 
+        c.open();
+        credInDomain = c.get(UserPwdCredential.class, domainName, CRED_ID);
+        credInDomain.delete.click();
+        c.save();
+        
+        // verify credential is not present
+        verifyCredentialNotPresent(domainName, CRED_ID);
+    }
+
+    /**
+     * Verifies that a given input for a given credential id 
+     * is equals to the value expected. If there are two credentials
+     * with the same id (that should not happen), the first one is verified.
```

Comment:
Well, basically the same reason mentioned in another comment. I believe that for the test to be meaningful, the page should be revisited and the value checked. Also, it is basically what was being done before my changes. See [here](https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/core/CredentialsTest.java#L34)